WordPress/five-for-the-future
# Manage Pledge: Update email dialog to use JS submission

Use the proper dialog behavior for submitting an email to generate a manage link.

Fixes #98.

When the "edit" link is clicked, the modal opens, and all other elements on the page are inactive. This uses [the `wicg-inert` polyfill](https://www.npmjs.com/package/wicg-inert) for `inert` support. Clicking anywhere on the page, or the X in the dialog, or hitting escape will close this.

![open](https://user-images.githubusercontent.com/541093/69287148-3ee5d780-0bc3-11ea-9c80-4161a96031e8.png)

When the correct email is submitted, the dialog closes, and shows this message (this is also announced using `wp.a11y.speak`).

![success](https://user-images.githubusercontent.com/541093/69287149-3ee5d780-0bc3-11ea-93ad-0e3b74c01ba0.png)

If the email does not match, the server responds with an error message we display in the dialog (and again announced using `wp.a11y.speak`):

![wrong](https://user-images.githubusercontent.com/541093/69287150-3ee5d780-0bc3-11ea-8de4-262357a5454d.png)

**To test**

- Go to a pledge
- Click "edit pledge"
- Enter a few emails, incorrect, correct, etc.
- With a correct email, you should get the "Updating your Pledge" email

cc @melchoyce for a design review.
